core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -16701,7 +16701,11 @@ public boolean isGroupExist(String groupID) throws UserStoreException {
             Object object = callSecure("isGroupExist", new Object[]{groupID}, argTypes);
             return (boolean) object;
         }
-        return StringUtils.isNotBlank(getGroupNameByGroupId(groupID));
+        try {
+            return StringUtils.isNotBlank(getGroupNameByGroupId(groupID));
```

Comment:
I think we should change the method instead of using `getGroupNameByGroupId`. Catching the exception and returning false doesn't exactly answer `isGroupExists`?

Reply:
The implementation is changed to check the error code and return false if group does not exist. Otherwise delegate the exception.

Reply:
Fixed by - https://github.com/wso2/carbon-kernel/pull/3364/commits/09471bb4bce3776ea82f3b4b8da75c80e4d8f916